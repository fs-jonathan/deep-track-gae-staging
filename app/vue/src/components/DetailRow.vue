<template>
  <div class="bg-white hover:bg-grey-light m-3 max-w-sm shadow-md rounded-md overflow-hidden">
    <div class="text-left p-1 sm:text-left sm:flex-grow">
      <span class="text-sm leading-tight">{{ this.getTitle(result.id) }}</span>
      <span class="text-xl leading-tight" style="float:right" v-if="result.id <= 8">¥{{ result.cost.toFixed(0) }}</span>
      <span class="text-xl leading-tight" style="float:right" v-else>{{ result.cost.toFixed(2) }}%</span>
    </div>
  </div>
</template>

<script>
export default {
  name: 'DetailRow',
  props: ['result'],
  methods: {
    getTitle: function(id) {
      switch (id) {
        case 6: return "見積もり収益額";
        case 7: return "ページビュー";
        case 8: return "表示回数";
        case 9: return "ページCTR";
        case 10: return "クリック率";
        case 11: return "カバレッジ";
        default: return "";
      }
    }
  }
}
</script>
