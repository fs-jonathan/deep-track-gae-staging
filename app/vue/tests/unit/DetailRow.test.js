import { shallowMount } from '@vue/test-utils'
import DetailRow from '@/components/DetailRow.vue'

describe('DetailRow.vue', () => {
  it('renders correctly', () => {
    const msg = 'new message'
    const wrapper = shallowMount(DetailRow, {
      propsData: { result: {
        title: msg,
        cost: 0,
        id: 2
      } }
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})
