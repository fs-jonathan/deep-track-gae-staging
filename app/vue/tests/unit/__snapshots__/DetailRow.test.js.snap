// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailRow.vue renders correctly 1`] = `
<div
  class="bg-white hover:bg-grey-light m-3 max-w-sm shadow-md rounded-md overflow-hidden"
>
  <div
    class="text-left p-1 sm:text-left sm:flex-grow"
  >
    <span
      class="text-sm leading-tight"
    >
      new message
    </span>
     
    <span
      class="text-xl leading-tight"
      style="float: right;"
    >
      Â¥0
    </span>
  </div>
</div>
`;
